src/main/java/co/cask/hydrator/plugin/BatchDynamoDBSink.java

```diff
@@ -0,0 +1,234 @@
+    private String endpointUrl;
+
+    @Nullable
+    @Description("The region for AWS Dynamo DB to connect to. Default is us-west-2 i.e. US West (Oregon).")
+    private String regionId;
+
+    @Macro
+    @Description("The access Id provided by AWS required to access the DynamoDb tables. (Macro Enabled)")
+    private String accessKey;
+
+    @Macro
+    @Description("AWS access key secret having access to DynamoDb tables. (Macro Enabled)")
+    private String secretAccessKey;
+
+    @Description("The table to write the data to. If the specified table does not exists, it will be created using " +
+      "the primary key attributes and key schema and the read and write capacity units.")
+    private String tableName;
+
+    @Description("A comma-separated list of key-value pairs representing the primary key and its attribute type. " +
+      "The value can be of the following type: 'N'(number), 'S'(string) and 'B'(binary - the byte[] value received " +
+      "from the previous stage will be converted to binary when storing the data in DynamoDB). Eg. \"Id:N,Region:S\"")
+    private String primaryKeyFields;
+
+    @Description("A comma-separated list of key-value pairs representing the primary key and its attribute type.\n" +
+      "The key type can have the following values: 'HASH' or 'RANGE'. Eg. \"Id:HASH,Region:HASH\".")
+    private String primaryKeyTypes;
+
+    @Nullable
+    @Description("The maximum number of strongly consistent reads consumed per second before DynamoDB returns a " +
+      "ThrottlingException. This will be used when creating a new table if the table name specified by the user does " +
+      "not exists.")
+    private String readCapacityUnits;
+
+    @Nullable
+    @Description("The maximum number of writes consumed per second before DynamoDB returns a ThrottlingException. " +
+      "This will be used when creating a new table if the table name specified by the user does not exists.")
+    private String writeCapacityUnits;
+
+    public DynamoDBConfig(String referenceName, @Nullable String endpointUrl, @Nullable String regionId,
+                          String accessKey, String secretAccessKey, String tableName, String primaryKeyFields,
+                          String primaryKeyTypes, @Nullable String readCapacityUnits,
+                          @Nullable String writeCapacityUnits) {
+      super(referenceName);
+      this.endpointUrl = endpointUrl;
+      this.regionId = regionId;
+      this.accessKey = accessKey;
+      this.secretAccessKey = secretAccessKey;
+      this.tableName = tableName;
+      this.primaryKeyFields = primaryKeyFields;
+      this.primaryKeyTypes = primaryKeyTypes;
+      this.readCapacityUnits = readCapacityUnits;
+      this.writeCapacityUnits = writeCapacityUnits;
+    }
+
+    /**
+     * Validates whether the table name follows the DynamoDB naming rules and conventions or not.
+     */
+    private void validateTableName() {
+      int tableNameLength = tableName.length();
```

Comment:
Agreed. Client creation and validation use the same code as DynamoDBSource plugin. 
Should we delete this and create a new repo? Or complete review for both plugins individually and then create and push re-factored code for both in the new repo?